cpp/example_code/rds/getting_started_with_db_instances.cpp

```diff
@@ -0,0 +1,1079 @@
+                      << std::endl;
+        }
+        // snippet-end:[cpp.example_code.rds.ModifyDBParameterGroup]
+    }
+
+    std::cout
+            << "You can get a list of parameters you've set by specifying a source of 'user'."
+            << std::endl;
+
+    Aws::Vector<Aws::RDS::Model::Parameter> userParameters;
+    // 6.  Display the modified parameters in the group
+    if (!getDBParameters(PARAMETER_GROUP_NAME, NO_NAME_PREFIX, "user", userParameters,
+                         client)) {
+        cleanUpResources(PARAMETER_GROUP_NAME, "", client);
+        return false;
+    }
+
+    for (const auto &userParameter: userParameters) {
+        std::cout << "  " << userParameter.GetParameterName() << ", " <<
+                  userParameter.GetDescription() << ", parameter value - "
+                  << userParameter.GetParameterValue() << std::endl;
+    }
+
+    printAsterisksLine();
+    std::cout << "Checking for an existing DB instance." << std::endl;
+
+    Aws::RDS::Model::DBInstance dbInstance;
+    // 7.  Check if the DB instance already exists.
+    if (!describeDBInstance(DB_INSTANCE_IDENTIFIER, dbInstance, client)) {
+        cleanUpResources(PARAMETER_GROUP_NAME, "", client);
+        return false;
+    }
+
+    if (dbInstance.DbInstancePortHasBeenSet()) {
+        std::cout << "The DB instance already exists." << std::endl;
+    }
+    else {
+        std::cout << "Let's create a DB instance." << std::endl;
+        const Aws::String administratorName = askQuestion(
+                "Enter an administrator username for the database: ");
+        const Aws::String administratorPassword = askQuestion(
+                "Enter a password for the administrator (at least 8 characters): ");
+        Aws::Vector<Aws::RDS::Model::DBEngineVersion> engineVersions;
+
+        // 8.  Get a list of available engine versions.
+        if (!getDBEngineVersions(DB_ENGINE, dbParameterGroupFamily, engineVersions,
+                                 client)) {
+            cleanUpResources(PARAMETER_GROUP_NAME, "", client);
+            return false;
+        }
+
+        std::cout << "The available engines for your parameter group are:" << std::endl;
+
+        int index = 1;
+        for (const Aws::RDS::Model::DBEngineVersion &engineVersion: engineVersions) {
+            std::cout << "  " << index << ": " << engineVersion.GetEngineVersion()
+                      << std::endl;
+            ++index;
+        }
+        int choice = askQuestionForIntRange("Which engine do you want to use? ", 1,
+                                            static_cast<int>(engineVersions.size()));
+        const Aws::RDS::Model::DBEngineVersion engineVersion = engineVersions[choice -
+                                                                              1];
+
+        Aws::String dbInstanceClass;
+        // 9.  Get a list of micro instance classes.
+        if (!chooseMicroDBInstanceClass(engineVersion.GetEngine(),
```

Comment:
Same thing here. Optional, but I did not understand why "micro" was in the snippet until I read it here. I'm not sure if a comment or another snippet is better, but I did feel like I was missing context from the single action example. It is fine (for me) in the scenario tags.